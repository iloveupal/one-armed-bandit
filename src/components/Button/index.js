import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';

const ButtonContainer = styled.div`
    width: 200px;
    height: 45px;
    border-radius: 5px;
    background-color: ${props => (props.secondary) ? '#fff' : '#2d3436'};
    color: ${props => (props.secondary) ? '#2d3436' : '#fff'};
    opacity: ${props => (props.isDisabled) ? 0.5 : 1};
`;


export default class Button extends PureComponent {
    static propTypes = {
        title: PropTypes.node.isRequired,
        isDisabled: PropTypes.bool,
        secondary: PropTypes.bool,
        onClick: PropTypes.func.isRequired,
    };

    handleClick = () => {
        if ( !this.props.isDisabled ) {
            this.props.onClick();
        }
    }

    render () {
        return (
            <ButtonContainer
                isDisabled={this.props.isDisabled}
                secondary={this.props.secondary}
                onClick={this.handleClick}
            >
                <span>{ this.props.title }</span>
            </ButtonContainer>
        );
    }
}
