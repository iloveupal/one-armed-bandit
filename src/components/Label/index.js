import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';

const LabelContainer = styled.div`
    width: 500px;
    height: 30px;
    border-radius: 5px;
    background-color: #27ae60;
    color: white;
    display: flex;
    flex-flow: row no-wrap;
    justify-content: center;
    align-items: center;
    margin: 0 15px;
    font-family: -apple-system, BlinkMacSystemFont, sans-serif;
    font-weight: bold;
`;


export default class Label extends PureComponent {
    static propTypes = {
        title: PropTypes.node.isRequired,
    };

    render () {
        return (
            <LabelContainer>
                <span>{ this.props.title }</span>
            </LabelContainer>
        );
    }
}
