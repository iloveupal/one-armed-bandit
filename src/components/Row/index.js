import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';


const RowDiv = styled.div`
    display: flex;
    flex-flow: row no-wrap;
    justify-content: center;
    padding: 20px 0;
`;


export default class Row extends PureComponent {
    static propTypes = {
        children: PropTypes.node.isRequired,
    };

    render () {
        return (
            <RowDiv>
                { this.props.children }
            </RowDiv>
        );
    }
}