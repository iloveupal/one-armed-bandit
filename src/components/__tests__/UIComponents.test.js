import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Button from '../Button';
import Label from '../Label';
import Row from '../Row';


describe('Ui Components', () => {
    test('Button component should match snapshot', () => {
        const component = renderer.create(
            <Button
                title="test"
                onClick={jest.fn()}
            />
        );

        expect(component.toJSON()).toMatchSnapshot();

        component.update(
            <Button
                title="test"
                onClick={jest.fn()}
                isDisabled
            />
        );

        expect(component.toJSON()).toMatchSnapshot();

        component.update(
            <Button
                title="test"
                onClick={jest.fn()}
                secondary
            />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Label component should match snapshot', () => {
        const component = renderer.create(
            <Label
                title="test"
            />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Row component should match snapshot', () => {
        const component = renderer.create(
            <Row>
                <div />
                <div />
                <div />
            </Row>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
});