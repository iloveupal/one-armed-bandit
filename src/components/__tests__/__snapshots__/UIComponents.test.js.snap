// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ui Components Button component should match snapshot 1`] = `
.c0 {
  width: 200px;
  height: 45px;
  border-radius: 5px;
  background-color: #2d3436;
  color: #fff;
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row no-wrap;
  -ms-flex-flow: row no-wrap;
  flex-flow: row no-wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 2px solid #2d3436;
  margin: 0 15px;
  font-family: -apple-system,BlinkMacSystemFont,sans-serif;
  font-weight: bold;
  cursor: pointer;
}

<div
  className="c0"
  onClick={[Function]}
>
  <span>
    test
  </span>
</div>
`;

exports[`Ui Components Button component should match snapshot 2`] = `
.c0 {
  width: 200px;
  height: 45px;
  border-radius: 5px;
  background-color: #2d3436;
  color: #fff;
  opacity: 0.5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row no-wrap;
  -ms-flex-flow: row no-wrap;
  flex-flow: row no-wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 2px solid #2d3436;
  margin: 0 15px;
  font-family: -apple-system,BlinkMacSystemFont,sans-serif;
  font-weight: bold;
  cursor: pointer;
}

<div
  className="c0"
  onClick={[Function]}
>
  <span>
    test
  </span>
</div>
`;

exports[`Ui Components Button component should match snapshot 3`] = `
.c0 {
  width: 200px;
  height: 45px;
  border-radius: 5px;
  background-color: #fff;
  color: #2d3436;
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row no-wrap;
  -ms-flex-flow: row no-wrap;
  flex-flow: row no-wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 2px solid #2d3436;
  margin: 0 15px;
  font-family: -apple-system,BlinkMacSystemFont,sans-serif;
  font-weight: bold;
  cursor: pointer;
}

<div
  className="c0"
  onClick={[Function]}
>
  <span>
    test
  </span>
</div>
`;

exports[`Ui Components Label component should match snapshot 1`] = `
.c0 {
  width: 500px;
  height: 30px;
  border-radius: 5px;
  background-color: #27ae60;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row no-wrap;
  -ms-flex-flow: row no-wrap;
  flex-flow: row no-wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 15px;
  font-family: -apple-system,BlinkMacSystemFont,sans-serif;
  font-weight: bold;
}

<div
  className="c0"
>
  <span>
    test
  </span>
</div>
`;

exports[`Ui Components Row component should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row no-wrap;
  -ms-flex-flow: row no-wrap;
  flex-flow: row no-wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 20px 0;
}

<div
  className="c0"
>
  <div />
  <div />
  <div />
</div>
`;
